ocaml/xapi/extauth_plugin_ADwinbind.ml

```diff
@@ -105,7 +192,45 @@ module AuthADWinbind : Auth_signature.AUTH_MODULE = struct
           explicitly filter any one-time credentials [like AD username/password for example] that it
           does not need long-term.]
   *)
-  let on_enable config_params = debug "on_enable To be implemented in CP-36085"
+  let on_enable config_params =
+    let user =
+      from_config ~name:"user" ~err_msg:"enable requires username"
+        ~config_params
+    in
+    let pass =
+      from_config ~name:"pass" ~err_msg:"enable requires password"
+        ~config_params
+    in
+
+    assert_hostname_valid () ;
+
+    let service_name =
+      (fun __context ->
+        Helpers.get_localhost ~__context |> fun host ->
+        Db.Host.get_external_auth_service_name ~__context ~self:host)
+      |> Server_helpers.exec_with_new_task
```

Comment:
what's the rationale for creating a new task here? (particularly for such a small piece of work)

Reply:
I port such code pieces from PBIS, 
for me, just want to refer the argument `__context` from new task, 
is there any other way I can get `__context` argument in this function?

Reply:
ah i see, it is strange the existing interface does not provide you with a context. There isn't another way really

Personally I would wrap the whole `on_enable` function in a task, and call it `on_enable`, but this is fine